src/Clerk.ts

```diff
@@ -276,6 +285,50 @@ export default class Clerk {
     });
   }
 
+  withProtectedRouteSSR(destination: string){
+    return (handler?: (ctxWithSession: SSRPropsContext) => Promise<GetServerSidePropsResult<any>>) => async (ctx: GetServerSidePropsContext) => {
+
+      // GetServerSidePropsContext req does not contain the query.
+      // For the sake of keeping the abstraction cleaner we inject it from ctx
+      // @ts-ignore
+      ctx.req.query = ctx.query;
+
+      try{
+        await this._runMiddleware(ctx.req, ctx.res, this.expressWithSession({ onError: this.strictOnError }));        
+      }catch(err){
+        return {
+          redirect: {
+            destination,
+            permanent: false,
+          },
+        }
+      }
+
+      return handler ? (await handler(ctx as SSRPropsContext)) : {props: {}};
+    }
+  }
+
+  withSessionSSR(handler: (ctxWithSession: SSRPropsContext) => Promise<GetServerSidePropsResult<any>>, { onError }: MiddlewareOptions = { onError: this.defaultOnError }) {
+    return async (ctx: GetServerSidePropsContext) => {
+      // GetServerSidePropsContext req does not contain the query.
+      // For the sake of keeping the abstraction cleaner we inject it from ctx
+      // @ts-ignore
+      ctx.req.query = ctx.query;
+
+      try {
+        await this._runMiddleware(ctx.req, ctx.res, this.expressWithSession({ onError }));
+        return await handler(ctx as SSRPropsContext);
+      } catch (error) {
+        ctx.res.statusCode = error.statusCode || 401;
```

Comment:
❓ Is this code reachable if the onError middleware option is triggered?

Reply:
@SokratisVidros Thanks for the question. Yes this code is reached if the following conditions are true:
1. The user has passed the `onError` property as a function.
2. The `onError` function returns an `Error` object as the `next` middleware handler expects.
Do you propose any changes to this behaviour here ?

Reply:
Nope, all good.